pkg/controllers/workgenerator/controller.go

```diff
@@ -1530,16 +1530,30 @@ func (r *Reconciler) SetupWithManager(mgr controllerruntime.Manager) error {
 					newResourceSnapshot := newWork.Labels[fleetv1beta1.ParentResourceSnapshotIndexLabel]
 					if oldResourceSnapshot == "" || newResourceSnapshot == "" {
 						klog.ErrorS(controller.NewUnexpectedBehaviorError(errors.New("found an invalid work without parent-resource-snapshot-index")),
-							"Could not find the parent resource snapshot index label", "oldWork", klog.KObj(oldWork), "oldResourceSnapshotLabelValue", oldResourceSnapshot,
-							"newWork", klog.KObj(newWork), "newResourceSnapshotLabelValue", newResourceSnapshot)
+							"Could not find the parent resource snapshot index label", "oldWork", klog.KObj(oldWork), "oldWorkLabels", oldWork.Labels,
+							"newWork", klog.KObj(newWork), "newWorkLabels", newWork.Labels)
 						return
 					}
-					// There is an edge case that, the work spec is the same but from different resourceSnapshots.
-					// WorkGenerator will update the work because of the label changes, but the generation is the same.
+					oldClusterResourceOverrideSnapshotHash := oldWork.Annotations[fleetv1beta1.ParentClusterResourceOverrideSnapshotHashAnnotation]
```

Comment:
keep the comment for the resourceSnapshot?

Reply:
it's moved to 1549